quic/client/handshake/FizzClientContext.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#pragma once
+
+#include <quic/client/handshake/ClientHandshakeFactory.h>
+
+namespace quic {
+
+class FizzClientHandshake;
+
+class FizzClientContext
+    : public ClientHandshakeFactory,
+      public std::enable_shared_from_this<FizzClientContext> {
```

Comment:
is there a reason this class itself needs to be an enable_shared from this? do you plan on creating shared ptrs from inside this class?

Reply:
Yes. The Handshake object will need to have a reference to it. See for instance #63 . Because it is the factory, it needs to pass itself down to the handshake it creates.

Reply:
i see, its fine then